lib/reek/ast/object_refs.rb

```diff
@@ -1,33 +1,30 @@
 module Reek
-  module AST
+  module AST  # :nodoc:
+    ObjectRef = Struct.new(:name)
     #
     # Manages and counts the references out of a method to other objects.
     #
     # @api private
     class ObjectRefs  # :nodoc:
       def initialize
-        @refs = Hash.new(0)
+        @refs = Hash.new { |refs, name| refs[name] = [] }
       end
 
-      def record_reference_to(exp)
-        @refs[exp] += 1
+      def most_popular
+        max = @refs.values.map(&:size).max
+        @refs.select { |_name, refs| refs.size == max }
```

Comment:
Yeah, it was super irritating. Went through all of the `enumerable` docs (always a good thing!) to make sure there isn’t a better way. :)